### PR TITLE
Implement hubs feature on the backend

### PR DESCRIPTION
**Pull request by ryzokuken**

### CODE REVIEW

File: api/hub_api.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+	if hub.ID == 0 {
+		response := make(map[string]interface{})
+		response["status"] = "FAILED"
+		response["error"] = "Hub not found"
+		return c.JSON(http.StatusNotFound, response)
+	}
+
+	response := make(map[string]interface{})
+	// TODO: Write Serializers
```
Comment (yashmehrotra) [inline api/hub_api.go]: @ryzokuken Please write the serializers :stuck_out_tongue:

File: api/hub_api.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		response["status"] = "FAILED"
+		response["error"] = "Hub with similar slug already exists"
+		return c.JSON(http.StatusConflict, response)
+	}
+
+	response := utils.SuccessResponse()
+	return c.JSON(http.StatusCreated, response)
+}
+
+func ShowHub(c echo.Context) error {
```
Comment (yashmehrotra) [inline api/hub_api.go]: func GetHub

File: server.go [hunk 3]
```diff
@@ -17,11 +17,18 @@ func main() {
...
 	e.POST("/api/user/new", api.CreateUser)
-	e.POST("/api/user/webkioskauth", api.WebkioskAuth)
+	// e.POST("/api/user/webkioskauth", api.WebkioskAuth)
 	e.POST("/api/user/login", api.LoginUser)
-	//e.GET("/insert_hub", api.Insert_hub)
-	//e.GET("/insert_event", api.Insert_event)
+
+	// Hub Actions
+	e.POST("/api/hub/new", api.CreateHub)
+	e.GET("/api/hub/show", api.ShowHub)
```
Comment (yashmehrotra) [inline server.go]: /api/hub/:slug

File: api/hub_api.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	var hub models.Hub
+
+	payload := utils.ParseJSON(c.Request().Body)
+
+	slug := payload["slug"]
```
Comment (yashmehrotra) [inline api/hub_api.go]: slug := c.Param("slug")

Comment (yashmehrotra) [reply api/hub_api.go]: If no slug, return all hubs

**Review by yashmehrotra — CHANGES_REQUESTED** [review]

File: server.go (hunk 3 above)
Comment (ryzokuken) [reply server.go]: we'd make it that way on the frontend. Why use URL params on the backend?

File: server.go [hunk 5]
```diff
@@ -17,11 +17,18 @@ func main() {
...
-	e.POST("/api/user/webkioskauth", api.WebkioskAuth)
+	// e.POST("/api/user/webkioskauth", api.WebkioskAuth)
 	e.POST("/api/user/login", api.LoginUser)
-	//e.GET("/insert_hub", api.Insert_hub)
-	//e.GET("/insert_event", api.Insert_event)
+
+	// Hub Actions
+	e.POST("/api/hub/new", api.CreateHub)
+	e.GET("/api/hub/:slug", api.GetHub)
+	e.GET("/api/hub", api.ShowHubs)
```
Comment (yashmehrotra) [inline server.go]: GetAllHubs would be more consistent

**Review by yashmehrotra — CHANGES_REQUESTED** [review]
1. Serializers should be in their respective models.
2. Please Gofmt the files.
3. Name them like, user.Serialize() etc.

File: models/hub.go [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import "github.com/jinzhu/gorm"
+
+type Hub struct {
+	gorm.Model
+	Name  string
+	About string
+	Slug  string `gorm:"not null;unique"`
+}
+
+func (hub Hub) SerializeHub() map[string]interface{} {
```
Comment (yashmehrotra) [inline models/hub.go]: Just `Serialize`

File: models/user.go [hunk 7]
```diff
@@ -20,3 +20,14 @@ func (user *User) BeforeCreate(scope *gorm.Scope) error {
 	scope.SetColumn("Slug", "TODO")
 	return nil
 }
+
+func (user User) SerializeUser() map[string]interface{} {
```
Comment (yashmehrotra) [inline models/user.go]: Same, .... just `Serialize`

**Review by yashmehrotra — CHANGES_REQUESTED** [review]

**Pull request merged**